debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java

```diff
@@ -0,0 +1,540 @@
+        RELATION,
+        BEGIN,
+        COMMIT,
+        INSERT,
+        UPDATE,
+        DELETE,
+        TYPE,
+        ORIGIN;
+
+        public static MessageType forType(char type) {
+            switch (type) {
+                case 'R': return RELATION;
+                case 'B': return BEGIN;
+                case 'C': return COMMIT;
+                case 'I': return INSERT;
+                case 'U': return UPDATE;
+                case 'D': return DELETE;
+                case 'Y': return TYPE;
+                case 'O': return ORIGIN;
+                default: throw new IllegalArgumentException("Unsupported message type: " + type);
+            }
+        }
+    }
+
+    public PgOutputMessageDecoder(MessageDecoderConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public boolean shouldMessageBeSkipped(ByteBuffer buffer, Long lastReceivedLsn, Long startLsn) {
+        LOGGER.trace("==============================================================================");
+        LOGGER.trace("Start LSN: {}, Last Received: {}", startLsn, lastReceivedLsn);
+
+        // Cache position as we're going to peak at the first byte to determine message type
+        // We need to reprocess all BEGIN/COMMIT messages regardless.
+        int position = buffer.position();
+        try {
+            MessageType type = MessageType.forType((char) buffer.get());
+            LOGGER.trace("Message Type: {}", type);
+            switch(type) {
+                case BEGIN:
+                case COMMIT:
+                case RELATION:
+                    // BEGIN/COMMIT
+                    // These types should always be processed due to the nature that they provide
+                    // the stream with pertinent per-transaction boundary state we will need to
+                    // always cache as we potentially might reprocess the stream from an earlier
+                    // LSN point.
+                    //
+                    // RELATION
+                    // These messages are always sent with a lastReceivedLSN=0; and we need to
+                    // always accept these to keep per-stream table state cached properly.
+                    LOGGER.trace("{} messages are always reprocessed", type);
+                    return false;
+                default:
+                    // INSERT/UPDATE/DELETE/TYPE/ORIGIN
+                    // These should be excluded based on the normal behavior, delegating to default method
+                    return MessageDecoder.super.shouldMessageBeSkipped(buffer, lastReceivedLsn, startLsn);
+            }
+        }
+        finally {
+            // Reset buffer position
+            buffer.position(position);
+        }
+    }
+
+    @Override
+    public void processMessage(ByteBuffer buffer, ReplicationMessageProcessor processor, TypeRegistry typeRegistry) throws SQLException, InterruptedException {
+        final MessageType messageType = MessageType.forType((char) buffer.get());
```

Comment:
Could you please trace log a hex representation of the raw message before decoding? It was very imporatnt when we were debigging streaming wal2json issues on AWS - to see what's coming out of the database.

Reply:
Added.

Reply:
I think the message is binary, not textual so it should be encoded into string using `io.debezium.util.HexConverter.convertToHexString(byte[])`. Also this is expensive operation so it should be guraded using `isTraceEnabled()`